compiler/rustc_mir_transform/src/check_unsafety.rs

```diff
@@ -600,13 +622,46 @@ pub fn check_unsafety(tcx: TyCtxt<'_>, def_id: LocalDefId) {
 
                 err.emit();
             }
-            UnsafetyViolationKind::UnsafeFn => tcx.struct_span_lint_hir(
-                UNSAFE_OP_IN_UNSAFE_FN,
-                lint_root,
-                source_info.span,
-                format!("{} is unsafe and requires unsafe block (error E0133)", description,),
-                |lint| lint.span_label(source_info.span, description).note(note),
-            ),
+            UnsafetyViolationKind::UnsafeFn => match details {
```

Comment:
Same here, the three cases should ideally be unified: they are all the same except for what's inside the closure. Ideally, one would just always call `the note_missing_feature` function. Alternatively, one can also remove the `CallToFunctionWith` matching inside `note_missing_feature`. I just think that matching twice makes no sense at all.